docs/src/api.md

```diff
@@ -1,30 +1,76 @@
 # API
 
-```@eval
-using NLPModels
-using Base.Markdown
-
-s = []
-mtds = [obj, grad, grad!, cons, cons!, jac_coord, jac, jprod, jprod!,
-  jtprod, jtprod!, hess_coord, hess, hprod, hprod!, NLPtoMPB, reset!]
-models = [("SimpleNLPModel","simple_model.jl"),
-  ("JuMPNLPModel","jump_model.jl"), ("SlackModel","slack_model.jl")]
-path = Pkg.dir("NLPModels", "src")
-files = map(x->readall(open(joinpath(path, x[2]))), models)
-
-for i = 1:length(mtds)
-  name = split(string(mtds[i]), ".")[2]
-  push!(s, md"### $name")
-  push!(s, @doc mtds[i])
-  sout = []
-  for j = 1:length(files)
-    if contains(files[j], "function $name")
-      md = models[j][1]
-      push!(sout, "[$md](/models/#$md)")
-    end
-  end
-  push!(s, "Implemented by " * join(sout, ", "))
-end
-
-s
+As stated in the [home](home) page, we consider the nonlinear optimization
+problem in the following format:
+\begin{align*}
+\min \quad & f(x) \\\\
+& c_L \leq c(x) \leq c_U \\\\
+& \ell \leq x \leq u.
+\end{align*}
+To develop an optimization algorithm, we are usually worried not only with
+$f(x)$ and $c(x)$, but also with their derivatives.
+Namely,
+
+- $\nabla f(x)$, the gradient of $f$ at the point $x$;
+- $\nabla^2 f(x)$, the Hessian of $f$ at the point $x$;
+- $J(x) = \nabla c(x)$, the Jacobian of $c$ at the point $x$;
+- $\nabla^2 f(x) + \sum_{i=1}^m \lambda_i \nabla^2 c_i(x)$,
+  the Hessian of the Lagrangian function at the point $(x,\lambda)$.
+
+There are many ways to access some of these values, so here is a little
+reference guide.
+
+## Reference guide
+
+The following naming should be easy enough to follow.
+If not, click on the link and go to the description.
+
+- `!` means inplace;
+- `_coord` means coordinate format;
+- `prod` means matrix-vector product;
+- `_op` means operator (as in [LinearOperators.jl](https://github.com/JuliaSmoothOptimizers/LinearOperators.jl)).
+
+If there is some operator you feel should be default for every model, open an
+issue, so we can discuss.
```

Comment:
How about: Feel free to open an issue to suggest other methods that should apply to all `NLPModel` instances.